src/main/kotlin/teaforge/platform/RoboRio/internal/Subscriptions.kt

```diff
@@ -206,3 +228,25 @@ fun <TMessage, TModel> runReadHidPort(
 
         return Triple(model, state, Maybe.Some(state.config.message(hidValue)))
 }
+
+fun <TMessage, TModel> runReadInterval(
+        model: RoboRioModel<TMessage, TModel>,
+        state: SubscriptionState.Interval<TMessage>
+): Triple<RoboRioModel<TMessage, TModel>, SubscriptionState<TMessage>, Maybe<TMessage>>{
+        val currentMicroseconds = HALUtil.getFPGATime()
+        return if (currentMicroseconds >= state.nextReadTimeMicroseconds){
+                val intervalMicroseconds = state.config.millisecondsBetweenReads * 1_000L
+                val intervalsMissed = ((currentMicroseconds - state.nextReadTimeMicroseconds) / intervalMicroseconds) + 1
+
+                val newNextReadTime = state.nextReadTimeMicroseconds + intervalMicroseconds * intervalsMissed
+
+                val elapsedMilliseconds = (currentMicroseconds - state.nextReadTimeMicroseconds) / 1_000L
+
+                val updatedState = state.copy(nextReadTimeMicroseconds = newNextReadTime)
+                Triple(model, updatedState, Maybe.Some(state.config.message(elapsedMilliseconds)))
```

Comment:
If you do the change to use microseconds above then this changes to
```suggestion
                Triple(model, updatedState, Maybe.Some(state.config.message(elapsedMicroseconds / 1_000L)))
```